taskboard/types.py

```diff
@@ -0,0 +1,13 @@
+from dataclasses import dataclass
+from typing import Optional
+
+@dataclass
+class Task:
```

Comment:
add doc strings to all interface classes and methods.

Reply:
Done.